src/Java.Interop/Java.Interop/JniPeerMembers.JniInstanceMethods_Invoke.cs

```diff
@@ -50,8 +52,8 @@ public unsafe bool InvokeAbstractBooleanMethod (string encodedMember, IJavaPeera
 
 				var m   = GetMethodInfo (encodedMember);
 
-				return JniEnvironment.InstanceMethods.CallBooleanMethod (self.PeerReference, m, parameters);
-				
+				var r   = JniEnvironment.InstanceMethods.CallBooleanMethod (self.PeerReference, m, parameters);
```

Comment:
Looks like there's missing `GC.KeepAlive (self);` call.

Reply:
Or is it because `bool` type here? The change still looks weird, as well as others below.

Reply:
Good catch!  Looks like I missed some spots within `JniPeerMembers.JniInstanceMethods_Invoke.tt`.

Reply:
This has been addressed in the latest force-push.